src/shared/ui/TextField/TextField.tsx

```diff
@@ -0,0 +1,28 @@
+import { TextFieldContext, type TextFieldErrorType } from './TextFieldContext';
+import Fieldset from './Fieldset';
+import Label from './Label';
+import Button from '../Button/Button';
+import Input from './Input';
+import HelperText from './HelperText';
+
+const TextField = ({
+  id,
+  placeholder,
+  error,
+  children,
+}: {
+  id?: string;
+  placeholder?: string;
+  error?: TextFieldErrorType;
+  children: React.ReactNode;
+}) => {
+  return <TextFieldContext.Provider value={{ id, placeholder, error }}>{children}</TextFieldContext.Provider>;
+};
+
+TextField.Fieldset = Fieldset;
```

Comment:
Fieldset은 껍데기 역할+모든 상황에 필요하므로 하위 컴포넌트로 빼지 않고, 부모 컴포넌트 내에 넣어놔도 좋을 것 같아요!

```suggestion
  return (
    <TextFieldContext.Provider value={{ id, placeholder, error }}>
      <fieldset className="mb-6 flex flex-col gap-2 border-0 p-0">
        {children}
      </fieldset>
    </TextFieldContext.Provider>
  );
};
```